api/envoy/api/v2/core/BUILD

```diff
@@ -35,7 +35,7 @@ api_proto_library_internal(
     name = "base",
     srcs = ["base.proto"],
     visibility = [
-        ":friends",
+        "//visibility:public",
```

Comment:
why did you need to change this?

Reply:
Without the public visibility qualifier and given the bazel WORKSPACE extract shown below, the following error is encountered:
```
# Envoy API definitions

## We import envoy_api as a sub-repo of the mono-repo
git_repository(
    name = "envoy",
    branch = "data-plane",
    remote = "https://github.com/thales-e-security/envoy.git",
    verbose = True,
)

# Create @envoy_api virtual repository.
load("@envoy//bazel:api_repositories.bzl", "envoy_api_dependencies")

envoy_api_dependencies()

load("@envoy_api//bazel:repositories.bzl", "api_dependencies")

# api_dependencies() must be called first to import the correct versions of dependencies.
api_dependencies()
```

```
ERROR: xxx/.cache/bazel/_bazel_xxx/xxx/external/envoy_api/envoy/service/auth/v2/BUILD:5:1: in proto_library rule @envoy_api//envoy/service/auth/v2:attribute_context: target '@envoy_api//envoy/api/v2/core:base' is not visible from target '@envoy_api//envoy/service/auth/v2:attribute_context'. Check the visibility declaration of the former target if you think the dependency is legitimate
```

It appears https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/BUILD#L15 allows the external authz service definition to use types defined in base.proto. However, external implementers of https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/auth/v2/external_auth.proto are not afforded the same rights.

The options I considered were:

- Make the contents of base.proto public
- Move the required types into another proto file and make that public.

The chose the former for simplicity as well as not wanting to cause significant rework for existing filters.

Reply:
I cannot reproduce your error with bazel 0.27.0 with a project importing envoy like

```
local_repository(
    name = "envoy",
    path = "/Users/lizan/github/lizan/envoy/7459",
)


load("@envoy//bazel:api_repositories.bzl", "envoy_api_dependencies")

envoy_api_dependencies()

load("@envoy//bazel:repositories.bzl", "GO_VERSION", "envoy_dependencies")
load("@envoy//bazel:cc_configure.bzl", "cc_configure")

envoy_dependencies()

load("@rules_foreign_cc//:workspace_definitions.bzl", "rules_foreign_cc_dependencies")

rules_foreign_cc_dependencies()

cc_configure()

load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")

go_rules_dependencies()

go_register_toolchains(go_version = GO_VERSION)
```

The code in /Users/lizan/github/lizan/envoy/7459 contains this PR and reverted this line.

```
bazel build @envoy_api//envoy/service/auth/v2:external_auth_service_cc_grpc
```
works well too.

The visibility restriction is to avoid accidental unindented dependencies to core, so this shouldn't be changed.

Reply:
Hmmm. With bazel 0.25.2 and 0.27.1 on Linux I get the original error even with the WORKSPACE snippet provided. I'll dig more tomorrow.

Reply:
@lizan you were right. I will revert to a visibility of `:friends`.